### PR TITLE
feature: confirm re-design header component implementation

### LINKED ISSUE #21279: Creating reusable header component for confirmation pages
We need to creating reusable header component for confirmation pages.

Designs: https://www.figma.com/file/uACzbBjwU2tV30UuewYiTx/Confirmation-redesign-V2?type=design&node-id=775-28145&mode=design&t=fcwGIVZS3lkt9rOO-0

This can be placed at /ui/components/app/confirmations

For the data that is required by header component we need to create / use existing hooks.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBemdkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--055ad992e73e45f18866ac97b7df0287e07ca3fd/Screenshot%202023-10-10%20at%203.45.29%E2%80%AFPM.png)

### Out of scope
There will be separate tasks to implement both the info icon (the icon on the left) and advanced user toggle (the icon further to the right) in header component.
- Info icon issue: https://app.zenhub.com/workspaces/-confirmations-secure-ux-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-extension/21280

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

New header component for confirmation re-design.

## **Related issues**

Fixes: #21279 

## **Manual testing steps**

1. Enable `conf-redesign` in local main build
2. Go to Test DAPP
3. Submit personal signature request
4. Check new header on confirmation page

## **Screenshots/Recordings**
Screenshot 2023-11-29 at 4 13 15 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by segun — DISMISSED** [review]
Changes look good. Fix CI Failures.

**Review by segun — DISMISSED** [review]

**Review by digiwand — CHANGES_REQUESTED** [review]
added suggestions to update styles, classnames, and tests. I also included an open-ended thought related to storybook controls

File: ui/components/app/confirm/header/header.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.confirm_header {
+  &-wrapper {
+    box-shadow: 0 2px 16px 0 #0000001a;
+    height: 64px;
+  }
+
+  &-avatar_network {
+    margin-left: -12px;
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/confirm/header/header.scss:8]: Hey this is the small network icon on top of account icon, I think RTL will not effect this.

File: ui/components/app/confirm/header/header.tsx [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
+const Header = () => {
+  const { networkImageUrl, networkDisplayName } = useConfirmationNetworkInfo();
+  const { recipientAddress, recipientName } = useConfirmationRecipientInfo();
+
+  return (
+    <Box
+      alignItems={AlignItems.center}
+      display={Display.Flex}
+      padding={4}
+      className="confirm_header-wrapper"
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/confirm/header/header.tsx:28]: Updated.

File: ui/components/app/confirm/header/header.tsx [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+    >
+      <Box display={Display.Flex}>
+        <Identicon address={recipientAddress} diameter={32} />
+        <AvatarNetwork
+          src={networkImageUrl}
+          name={networkDisplayName}
+          size={AvatarNetworkSize.Xs}
+          className="confirm_header-avatar_network"
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/confirm/header/header.tsx:36]: Updated.

File: ui/hooks/confirm/useConfirmationNetworkInfo.test.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        ...mockState,
+        confirm: {
+          currentConfirmation: { id: '1', msgParams: {} },
+        },
+      },
+    );
+
+    expect(result.current.networkDisplayName).toBe('Goerli');
+  });
+});
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/hooks/confirm/useConfirmationNetworkInfo.test.ts:33]: 👍 I added test cases

File: ui/hooks/confirm/syncConfirmPath.ts [hunk 5]
```diff
@@ -15,16 +15,18 @@ const syncConfirmPath = () => {
...
   useEffect(() => {
     if (!currentConfirmation) {
       return;
     }
     if (paramsTransactionId !== currentConfirmation.id) {
-      history.replace(
-        `${CONFIRM_TRANSACTION_ROUTE}/${currentConfirmation.id}/${SIGNATURE_REQUEST_PATH}`,
-      );
+      if (currentConfirmation.msgParams) {
+        // url for all signature requests
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/hooks/confirm/syncConfirmPath.ts]: This `if` condition is actually needed by TS also.

File: ui/hooks/confirm/useConfirmationNetworkInfo.test.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import mockState from '../../../test/data/mock-state.json';
+import { renderHookWithProvider } from '../../../test/lib/render-helpers';
+import useConfirmationNetworkInfo from './useConfirmationNetworkInfo';
+
+describe('useConfirmationNetworkInfo', () => {
+  it('returns network display name when confirmation chainId is present', () => {
```
Comment (digiwand) [inline ui/hooks/confirm/useConfirmationNetworkInfo.test.ts:6]:
```suggestion
  it('returns display name and image when confirmation chainId is present', () => {
```

File: ui/hooks/confirm/useConfirmationNetworkInfo.test.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          currentConfirmation: { id: '1', chainId: '0x1' },
+        },
+      },
+    );
+
+    expect(result.current.networkDisplayName).toBe('Ethereum Mainnet');
+    expect(result.current.networkImageUrl).toBe('./images/eth_logo.png');
+  });
+
+  it('returns network display name of provider chainId when confirmation chainId is not present', () => {
```
Comment (digiwand) [inline ui/hooks/confirm/useConfirmationNetworkInfo.test.ts:21]:
```suggestion
  it('returns display name and image of provider chainId when confirmation chainId is not present', () => {
```

**Review by digiwand — DISMISSED** [review]
updates look good 

[outstanding discussion](https://github.com/MetaMask/metamask-extension/pull/22081/files#r1427471966) is a non-blocker

File: ui/components/app/confirm/header/header.scss [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  &__wrapper {
+    box-shadow: 0 2px 16px 0 #0000001a;
+    height: 64px;
+  }
+
+  &__avatar-network {
+    margin-left: -12px;
+    margin-top: -4px;
+  }
+}
```
Comment (digiwand) [inline ui/components/app/confirm/header/header.scss:11]:
```suggestion
}

```

Screenshot 2023-12-15 at 4 16 22 PM

there's a lint error here. maybe this suggestion would fix it

**Review by digiwand — APPROVED** [review]

File: ui/components/app/confirm/header/header.test.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        },
+      },
+    },
+  });
+
+  return renderWithProvider(<Header />, store);
+};
+
+describe('Header', () => {
+  it('should match snapshot', async () => {
```
Comment (segun) [inline ui/components/app/confirm/header/header.test.tsx:27]:
```suggestion
  it('should match snapshot', () => {
```

**Review by segun — APPROVED** [review]
PR looks good. Left a small nit.

**Pull request merged**